Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -259,6 +263,19 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
       }
     })
   }
+  
+  /// Obtain the path of the directory to write generated module interfaces.
+  private static func generatedInterfacesPath(overrideInterfacesPath: URL?) throws -> URL {
+    if let overridePath = overrideInterfacesPath {
+      try FileManager.default.createDirectory(at: overridePath, withIntermediateDirectories: true)
+      return overridePath
+    } else {
+      let tempFolderURL = URL(fileURLWithPath: NSTemporaryDirectory())
+      let genFolder = tempFolderURL.appendingPathComponent("GeneratedInterfaces", isDirectory: true)
+      try FileManager.default.createDirectory(at: genFolder, withIntermediateDirectories: true)
+      return genFolder
+    }
+  }
```

Comment:
What do you think about defaulting the generated interface path to a temporary directory in `sourcekit-lsp/main.swift`. If you do that, the documentation of the `--generated-interfaces-path` could mention that it defaults to a temporary directory, which makes the behavior cleaner IMO.